processing/src/main/java/io/druid/query/search/SearchQueryQueryToolChest.java

```diff
@@ -152,7 +153,18 @@ public SearchQueryQueryToolChest(
         final byte[] querySpecBytes = query.getQuery().getCacheKey();
         final byte[] granularityBytes = query.getGranularity().cacheKey();
 
-        final Set<String> dimensions = Sets.newTreeSet();
+        final Set<DimensionSpec> dimensions = Sets.newTreeSet(
+            Ordering.from(
+                new Comparator<DimensionSpec>()
+                {
+                  @Override
+                  public int compare(DimensionSpec o1, DimensionSpec o2)
+                  {
+                    return o1.getOutputName().compareTo(o2.getOutputName());
```

Comment:
multiple queries having different outputName but on the same column maps to same cache key and results in same intermediate results. 
CachingClusteredClientTest.testGroupByCachingRenamedAggs does similar testing for groupBy.